homework/src/index.js

```diff
@@ -30,18 +42,141 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
+  //this is an object creator
+  function RepositoryInfo(sourceObj) {
+    this.Description = sourceObj.description;
+    this.Forks = sourceObj.forks;
+    const dateOfUpdate = new Date(sourceObj.updated_at);
+    this.Updated = dateOfUpdate.toLocaleString();
+  }
+
+  //info-box of the page
+  function createInfoBox(repoData, container) {
+
+    container.innerHTML = "";
+
+    const info = new RepositoryInfo(repoData);
+
+    const infoTable = createAndAppend('table', container, { id: 'infoTable' });
+
+    const infoBody = createAndAppend('tbody', infoTable, { id: 'infoBody' });
+
+
+    const repoTr = createAndAppend('tr', infoBody, { id: 'repoTr', class: 'rows' });
+
+    createAndAppend('td', repoTr, { html: 'Repository:', class: 'keys' });
+
+    const repoTd = createAndAppend('td', repoTr, { id: 'repoTd', class: 'values' });
+
+    createAndAppend('a', repoTd, { href: repoData.html_url, target: '_blank', html: repoData.name });
+
+
+    Object.keys(info).forEach((key) => {
+      const value = info[key];
+      const tr = createAndAppend('tr', infoBody, { id: `${key}Tr`, class: 'rows' });
+      createAndAppend('td', tr, { html: key + ':', class: 'keys' });
+      createAndAppend('td', tr, { html: value, class: 'values' });
     });
   }
 
+  //contributors box of the page
+  async function createContributorsBox(repoData, container) {
+
+    container.innerHTML = "";
+
+    const contributorsURL = repoData.contributors_url;
+
+    createAndAppend('p', container, { html: 'Contributions' });
+
+    try {
+      const contributors = await fetchJSON(contributorsURL, container);
+
+      const contributorsTable = createAndAppend('table', container, { id: 'contributorsTable' });
+
+      const contributorsTbody = createAndAppend('tbody', contributorsTable, { id: 'contributorsTbody' });
+
+
+      contributors.forEach(contributor => {
+
+        const contLink = contributor.html_url;
+        const contImg = contributor.avatar_url;
+        const contName = contributor.login;
+        const contNum = contributor.contributions;
+
+        const contributorsTr = createAndAppend('tr', contributorsTbody, {
+          id: `${contributor['id']}`,
+          class: 'contributorsTr',
+          onclick: `window.open('${contLink}', '_blank');`,
+          onkeydown: `if(event.key === 'Enter'){window.open('${contLink}', '_blank');};`,
+          role: 'link',
+          tabindex: 0
+        });
```

Comment:
Creating a table for this list is not the best solution. For one thing, it is not ARIA-compliant. Unless you add some extra ARIA attribute a screen reader will identify this as a table and not as a list. You also need to do complicated things here to respond to mouse click and the keyboard. If instead you used a `ul` with `li` and embedded `a` tags all this would become simpler. This is what another student did in terms of HTML structure:

```html
<ul class="contributor-list">
  <li class="contributor-item">
    <a target="_blank" href="https://github.com/gijscor">
      <img class="contributor-avatar" src="https://avatars0.githubusercontent.com/u/15912395?v=4">
      <div class="contributor-data">
        <div class="contributor_name">gijscor</div>
        <div class="contributor_badge">12</div>
      </div>
    </a>
  </li>
  <li class="contributor-item">
    <a target="_blank" href="https://github.com/wouterkleijn">
      <img class="contributor-avatar" src="https://avatars1.githubusercontent.com/u/37488847?v=4">
      <div class="contributor-data">
        <div class="contributor_name">wouterkleijn</div>
        <div class="contributor_badge">4</div>
      </div>
    </a>
  </li>
</ul>
```

Reply:
Actually using ul-li elements would be simpler than the table element. However, I added some extra codes, therefore a user can navigate through all interactive elements using the keyboard (using the Tab and Enter keys). Pressing Enter on such an element should be equivalent to clicking the mouse. 
I did not add some aria codes about screen reader. Instead of replacing the table element by ul-li elements, is it enough if I add some aria codes about screen reader?

Reply:
I think you will find that anchor elements (`a`) are navigable with the keyboard (Enter, Tab). In general, if you use HTML elements that are appropriate for the semantic context you are almost guaranteed to be ARIA-compliant. Only when you deviate from that you need extra ARIA attributes on your elements to provide the semantic context. So, when you want to create a list, use `ul` or `ol`. If you use a `table` than you would need add ARIA roles to label the table as a list and each row as a list item.

Check section 2 of  the link below (taken from the HTML/CSS module) for some simple rules for ARIA-compliance.

https://w3c.github.io/using-aria/